enhancements/update/standalone-worker-surge.md

```diff
@@ -0,0 +1,295 @@
+
+One cost-effective approach to ensure capacity is called "surging". With a surge strategy, during an update, the platform
+is permitted to bring additional worker nodes online, to accommodate workloads being drained from existing nodes. After an 
+update concludes, the surged nodes are scaled down and the cluster resumes its steady state.
+
+HyperShfit Hosted Control Planes (HCP) already support the surge concept. HyperShift `NodePools`
+expose `maxUnavailable` and `maxSurge` as configurable options during updates: https://hypershift-docs.netlify.app/reference/api/#hypershift.openshift.io/v1beta1.RollingUpdate . 
+Unfortunately, standalone OpenShift, which uses `MachineConfigPools`, does not. To workaround this
+limitation for managed services customers, Service Delivery developed a custom Machine Update Operator (MUO)
+which can surge a standalone cluster during an update (see [reserved capacity feature](https://github.com/openshift/managed-upgrade-operator/blob/a56079fda6ab4088f350b05ed007896a4cabcd97/docs/faq.md)).
+
+### Preventing Other Stalled Drains
+
+As previously mentioned, there are other reasons that drains can stall. For example, `PodDisruptionBudgets` can
+be configured in such a way as to prevent pods from draining even if there is sufficient capacity for them
+to be rescheduled on other nodes. A powerful (though blunt) tool to prevent drain stalls is to limit the amount of time
+a drain operation is permitted to run before forcibly terminating pods and allowing an update to proceed. 
+`NodeDrainTimeout`, in HCP's `NodePools` allows users to configure this timeout.
+The Managed Update Operator also supports this feature with [`PDBForceDrainTimeout`](https://github.com/openshift/managed-upgrade-operator/blob/master/docs/faq.md).
+
+This enhancement includes adding `NodeDrainTimeout` to `MachineConfigPools` to provide this feature in standlone 
+cluster environments.
+
+### User Stories
+
+Implementing surge and node drain timeout support in `MachineConfigPools` can simplify cluster management for self-managed 
+standalone clusters as well as managed clusters (i.e. Service Delivery can remove this customized behavior from the MUO and use
+more of the core platform).
+
+* As an Operations team managing one or more standalone clusters, I want to
+  help ensure smooth updates by surging my worker node count without constantly 
+  having my cluster over-provisioned.
+* As an Operations team managing one or more standalone clusters, I want to 
+  ensure my cluster update makes steady progress by limiting the amount of time a node drain can
+  consume.
+* As an engineer in the Service Delivery organization, I want to use core platform
+  features instead of developing, evolving, and testing the Managed Update Operator.
+* As an Operations team managing standalone and HCP based OpenShift clusters, I
+  want a consistent update experience leveraging a surge strategy and/or 

```

Comment:
HCP does replace nodes by default, I believe you can opt for in-place updates also.  For managed openshift we use replace.  I'm +1 to getting the option for replace for classic if that's the direction OCP wants to go.  We do have some customers in HCP that require in-place upgrades if I remember correctly, so it needs to be configurable.

Reply:
As far as I'm concerned, yes, it is something we want to bring in. I have this on the discussion topics already for the CAPI F2F end of July where I'm hoping to get more consensus on how we are going to handle upgrades (CVO/MCO) in a more CAPI driven OpenShift land.

I imagine that yes, in the future, both HCP and OCP will support both, and it would be preferable I think to leverage the same building blocks to do so, rather than inventing something new